src/app/services/swagger.service.ts

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-console */
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { environment } from 'src/environments/environment';
+import { SwaggerSummary } from '../models/swagger-summary/swagger-summary';
+import { SwaggerUploadResponse } from '../models/swagger-upload-response/swagger-upload-response';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SwaggerService {
+  public uploadForm: FormGroup;
+
+  public secondsUntilETA: number;
+
+  public swaggerSummary: SwaggerSummary;
+
+  public swaggerUploadResponse: SwaggerUploadResponse;
+
+  constructor(private http: HttpClient) { }
+
+  async uploadSwaggerFile(formData: FormData): Promise <SwaggerUploadResponse> {
```

Comment:
Style wise, generics usually prefer not to have spaces between the wrapper type. So `Promise <SwaggerUploadResponse>` might be just `Promise<SwaggerUploadResponse>`.